primitives/runtime-interface/test/src/lib.rs

```diff
@@ -104,30 +110,52 @@ fn host_function_not_found() {
 		\\\"Invalid utf8 data provided\\\")) }\""
 )]
 fn test_invalid_utf8_data_should_return_an_error() {
-	call_wasm_method::<HostFunctions>("test_invalid_utf8_data_should_return_an_error");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_invalid_utf8_data_should_return_an_error");
 }
 
 #[test]
 fn test_overwrite_native_function_implementation() {
-	call_wasm_method::<HostFunctions>("test_overwrite_native_function_implementation");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_overwrite_native_function_implementation");
 }
 
 #[test]
 fn test_u128_i128_as_parameter_and_return_value() {
-	call_wasm_method::<HostFunctions>("test_u128_i128_as_parameter_and_return_value");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_u128_i128_as_parameter_and_return_value");
 }
 
 #[test]
 fn test_vec_return_value_memory_is_freed() {
-	call_wasm_method::<HostFunctions>("test_vec_return_value_memory_is_freed");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_vec_return_value_memory_is_freed");
 }
 
 #[test]
 fn test_encoded_return_value_memory_is_freed() {
-	call_wasm_method::<HostFunctions>("test_encoded_return_value_memory_is_freed");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_encoded_return_value_memory_is_freed");
 }
 
 #[test]
 fn test_array_return_value_memory_is_freed() {
-	call_wasm_method::<HostFunctions>("test_array_return_value_memory_is_freed");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_array_return_value_memory_is_freed");
+}
+
+#[test]
+fn test_verification_of_input() {
+	call_wasm_method::<HostFunctionsDeprecated>(
```

Comment:
```suggestion
	call_wasm_method::<HostFunctions>(
```

Not sure why we want to test the desecrated ones.

Reply:
We validate if they work with their own functions, just a sanity

Reply:
That is already done by all the other tests in here. Not required.

Reply:
Done not for deprecated wasm. Not strictly required, but still usefull. We test that deprecated wasm works as expected - not modified to the point where versioned test below would be useless to look at.

Reply:
You are not even testing the `test_verification_of_input_old` with the interface that provides `version(2)`.

We don't need to test the deprecated function with the interface that just exposes the deprecated interface. This doesn't test anything that isn't already tested by the other tests.

Reply:
It tests the fact that we can we have a right test setup to continue with testing further, just a sanity as I said initially